src/poketch_data.c

```diff
@@ -0,0 +1,314 @@
+        poketchData->unk_A4 = (1 << (param2 - 1));
+    }
+}
+
+void sub_02056970 (PoketchData *poketchData, u32 param1, u32 param2)
+{
+    if (poketchData->unk_A8 == param1) {
+        poketchData->unk_A4 &= ~(1 << (param2 - 1));
+    } else {
+        poketchData->unk_A8 = param1;
+        poketchData->unk_A4 = 0;
+    }
+}
+
+BOOL sub_020569A8 (const PoketchData *poketchData, u32 param1, u32 param2)
+{
+    if (poketchData->unk_A8 == param1) {
+        return (poketchData->unk_A4 >> (param2 - 1)) & 1;
+    } else {
+        return 0;
+    }
+}
+
+void PoketchData_SetMapMarker(PoketchData *poketchData, int index, u8 x, u8 y)
+{
+    GF_ASSERT(index < POKETCH_MAPMARKER_COUNT);
+
+    poketchData->markMapPositions[index].x = x;
+    poketchData->markMapPositions[index].y = y;
+}
+
+void PoketchData_MapMarkerPos(const PoketchData *poketchData, int index, u8 *x, u8 *y)
+{
+    GF_ASSERT(index < POKETCH_MAPMARKER_COUNT);
+
+    *x = poketchData->markMapPositions[index].x;
+    *y = poketchData->markMapPositions[index].y;
+}
+
+BOOL sub_02056A10 (const PoketchData *poketchData)
+{
+    return poketchData->unk_00_2;
+}
+
+void sub_02056A18 (const PoketchData *poketchData, u8 *param1)
+{
+    if (poketchData->unk_00_2) {
+        MI_CpuCopy8(poketchData->unk_2A, param1, 120);
+    }
+}
+
+void sub_02056A2C (PoketchData *poketchData, const u8 *param1)
+{
+    MI_CpuCopy8(param1, poketchData->unk_2A, 120);
+    poketchData->unk_00_2 = 1;
+}
+
+void sub_02056A48 (PoketchData *poketchData, const BoxPokemon * param1)
+{
+    int v0;
+
+    v0 = sub_02056AAC(poketchData);
+
+    if (v0 >= 12) {
+        int v1;
+
+        for (v1 = 0; v1 < (12 - 1); v1++) {
+            poketchData->unk_B8[v1] = poketchData->unk_B8[v1 + 1];
+        }
+
+        v0 = 12 - 1;
+    }
+
+    poketchData->unk_B8[v0].unk_00 = BoxPokemon_GetValue((BoxPokemon *)param1, MON_DATA_SPECIES, NULL);
+    poketchData->unk_B8[v0].unk_02 = BoxPokemon_IconFormOffset(param1);
+    poketchData->unk_B8[v0].unk_04 = BoxPokemon_GetValue((BoxPokemon *)param1, MON_DATA_FORM, NULL);
+}
+
+int sub_02056AAC (const PoketchData *poketchData)
+{
+    int v0;
+
+    for (v0 = 0; v0 < 12; v0++) {
+        if (poketchData->unk_B8[v0].unk_00 == 0) {
+            return v0;
+        }
+    }
+
+    return v0;
+}
+
+void sub_02056AC8 (const PoketchData *poketchData, int param1, int *param2, int *param3)
+{
+    GF_ASSERT(param1 < 12);
+    GF_ASSERT(poketchData->unk_B8[param1].unk_00);
+
+    *param2 = poketchData->unk_B8[param1].unk_00;
+    *param3 = poketchData->unk_B8[param1].unk_02;
+}
+
+u32 sub_02056AFC (const PoketchData *poketchData, int param1)
+{
+    GF_ASSERT(param1 < 12);
+    GF_ASSERT(poketchData->unk_B8[param1].unk_00);
+
+    return poketchData->unk_B8[param1].unk_04;
+}
+
+PoketchData *SaveData_PoketchData(SaveData *saveData)
+{
+    PoketchData *poketchData;
+
+    poketchData = SaveData_Get(saveData, SAVE_TABLE_ENTRY_POKETCH);
+    return poketchData;
```

Comment:
Does the following match?

```suggestion
    return SaveData_Get(saveData, SAVE_TABLE_ENTRY_POKETCH);
```